src/main/java/org/spongepowered/mod/mixin/world/MixinWorld.java

```diff
@@ -110,4 +115,73 @@ public Biome getBiome(Vector3d position) {
     public WorldBorder getWorldBorder() {
         return (WorldBorder) shadow$getWorldBorder();
     }
+
+    @Override
+    public Weather getWeather() {
+        if (this.worldInfo.isThundering()) {
+            return Weathers.THUNDER_STORM;
+        } else if (this.worldInfo.isRaining()) {
+            return Weathers.RAIN;
+        } else {
+            return Weathers.CLEAR;
+        }
+    }
+
+    @Override
+    public long getRemainingDuration() {
+        Weather weather = getWeather();
+        if (weather.equals(Weathers.CLEAR)) {
+            if (this.worldInfo.func_176133_A() > 0) {
+                return this.worldInfo.func_176133_A();
+            } else {
+                return Math.min(this.worldInfo.getThunderTime(), this.worldInfo.getRainTime());
+            }
+        } else if (weather.equals(Weathers.THUNDER_STORM)) {
+            return this.worldInfo.getThunderTime();
+        } else if (weather.equals(Weathers.RAIN)) {
+            return this.worldInfo.getRainTime();
+        }
+        return 0;
+    }
+
+    long weatherStartTime;
+
+    @Inject(method = "updateWeatherBody()V", at = { @At(value = "INVOKE", target = "setThundering(Z)V"), @At(value = "INVOKE", target = "setRaining(Z)V") })
+    private void onUpdateWeatherBody(CallbackInfo ci) {
+        this.weatherStartTime = this.worldInfo.getWorldTotalTime();
+    }
+
+    @Override
+    public long getRunningDuration() {
+        return this.worldInfo.getWorldTotalTime() - this.weatherStartTime;
+    }
+
+    @Override
+    public void forecast(Weather weather) {
+        this.forecast(weather, (300 + (new Random()).nextInt(600)) * 20);
```

Comment:
Is there anyway to hook into that code? Otherwise leaving a comment referencing where to find the code in 1.8 may help with future updates.